src/backends/policy_enforcement/souffle/value_test.cc

```diff
@@ -0,0 +1,172 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+
+#include "src/backends/policy_enforcement/souffle/value.h"
+
+#include <limits>
+
+#include "absl/strings/numbers.h"
+#include "absl/strings/string_view.h"
+#include "src/common/testing/gtest.h"
+
+namespace raksha::backends::policy_enforcement::souffle {
+
+using testing::Combine;
+using testing::TestWithParam;
+using testing::ValuesIn;
+
+class NumberTest : public TestWithParam<int64_t> {};
+
+TEST_P(NumberTest, NumberTest) {
+  int64_t num = GetParam();
+  Number number_value = Number(num);
+  std::string datalog_str = number_value.ToDatalogString();
+  int64_t parsed_int = 0;
+  bool conversion_succeeds = absl::SimpleAtoi(datalog_str, &parsed_int);
+  ASSERT_TRUE(conversion_succeeds);
```

Comment:
Make the string representation a part of the expectations and simply compare it against the return value of `number_value.ToDatalogString()`?

Reply:
That would make the test's logic a bit more concise, but as currently written, this allows us to have the input integer be also the expected value. This is especially nice for the max and min `int64_t` test inputs I'm using, as I don't know what the string representation for those is off the top of my head.